dune_client/interface.py

```diff
@@ -18,3 +18,16 @@ def refresh(self, query: Query) -> ResultsResponse:
         Executes a Dune query, waits till query execution completes,
         fetches and returns the results.
         """
+
+
+# pylint: disable=too-few-public-methods
+class AsyncDuneInterface(ABC):
```

Comment:
same as sync interface but has async interface for refresh?
also, may be it makes sense to add more functions to that interface?

Reply:
> same as sync interface but has async interface for refresh?
is this a question?

Yea, I think it would make plenty of sense to add more functions here. I am not exactly sure why I haven't. Did you have something particular in mind (apart from the obvious execute, get_status, cancel and get_results - which I agree are totally missing from here)? We could even remove that pylint disable!

Reply:
To be hounest I think I lack on the knowledge why is there an interface in the first place. So decided to let it be first, may be we can completely drop the interface and move all the logic to base class?

Reply:
Maybe we can keep the original (non async interface around because https://github.com/cowprotocol/dune-alerts depends on it (and it would be a breaking change), but perhaps there is no reason to introduce the async interface from the get go, and we can phase it out in a major version bump later.